packages/compiler-cli/ngcc/src/sourcemaps/source_file.ts

```diff
@@ -361,6 +356,33 @@ export function extractOriginalSegments(mappings: Mapping[]): Map<SourceFile, Se
   return originalSegments;
 }
 
+/**
+ * Update the original segments of each of the given `mappings` to include a link to the next
+ * segment in the source file.
+ *
+ * @param mappings the mappings whose segments should be updated
+ */
+export function ensureOriginalSegmentLinks(mappings: Mapping[]): void {
+  const segmentsBySource = extractOriginalSegments(mappings);
+  segmentsBySource.forEach(markers => {
+    for (let i = 0; i < markers.length - 1; i++) {
```

Comment:
By not iterating over all markers, will end up with a collection of markers of different shapes: Most have a `next` property and the last one does not. This will result to polymorphic calls afaict. (I doubt it will noticeably affect performance, but since it was discussed above...)

So, I suggest ensuring _all_ markers will have a `next` property (even it is is set to `undefined` or `null`).

```suggestion
    for (let i = 0; i < markers.length; i++) {
```

Reply:
The downside of going up to `markers.length` is that we end up doing a read on `markers[length]` which converts the array to unpacked.
Instead I will change the `SegmentMarker.next` to be `SegmentMarker|undefined` which means that the `next` property will have to be set (to `undefined`) at creation time so no need to update it here.